packages/hardhat-core/src/internal/core/config/config-resolution.ts

```diff
@@ -1,44 +1,242 @@
+            ])
+        )
+      : {};
+
+  return {
+    hardhat,
+    localhost,
+    ...otherNetworks,
+  };
+}
+
+function isHttpNetworkConfig(
+  config: UserNetworkConfig
+): config is UserHttpNetworkConfig {
+  return "url" in config;
+}
+
+function normalizeHexString(str: string): string {
+  const normalized = str.trim().toLowerCase();
+  if (normalized.startsWith("0x")) {
+    return normalized;
+  }
+
+  return `0x${normalized}`;
+}
+
+function resolveHardhatNetworkConfig(
+  hardhatNetworkConfig?: UserHardhatNetworkConfig
+): HardhatNetworkConfig {
+  if (hardhatNetworkConfig === undefined) {
+    hardhatNetworkConfig = {};
+  }
+
+  const clonedDefaultHardhatNetworkParams = cloneDeep(
+    defaultHardhatNetworkParams
+  );
+
+  const accounts: HardhatNetworkAccountConfig[] =
+    hardhatNetworkConfig.accounts === undefined
+      ? clonedDefaultHardhatNetworkParams.accounts
+      : Array.isArray(hardhatNetworkConfig.accounts)
+      ? hardhatNetworkConfig.accounts.map(({ privateKey, balance }) => ({
+          privateKey: normalizeHexString(privateKey),
+          balance,
+        }))
+      : normalizeHardhatNetworkAccountsConfig({
+          ...defaultHardhatNetworkHdAccountsConfigParams,
+          ...hardhatNetworkConfig.accounts,
+          mnemonic:
+            hardhatNetworkConfig.accounts.mnemonic ??
+            defaultHardhatNetworkHdAccountsConfigParams.menmonic,
+        });
+
+  const forking: HardhatNetworkForkingConfig | undefined =
+    hardhatNetworkConfig.forking !== undefined
+      ? {
+          url: hardhatNetworkConfig.forking.url,
+          enabled: hardhatNetworkConfig.forking.enabled ?? true,
+        }
+      : undefined;
+
+  const blockNumber = hardhatNetworkConfig?.forking?.blockNumber;
+  if (blockNumber !== undefined && forking !== undefined) {
+    forking.blockNumber = hardhatNetworkConfig?.forking?.blockNumber;
+  }
+
+  const config = {
+    ...clonedDefaultHardhatNetworkParams,
+    ...hardhatNetworkConfig,
+    accounts,
+    forking,
+  };
+
+  // We do it this way because ts gets lost otherwise
+  if (config.forking === undefined) {
+    delete config.forking;
+  }
+
+  return config;
+}
+
+function isHdAccountsConfig(
+  accounts: UserHttpNetworkAccountsConfig
+): accounts is UserHDAccountsConfig {
+  return typeof accounts === "object" && !Array.isArray(accounts);
+}
+
+function resolveHttpNetworkConfig(
+  networkConfig: UserHttpNetworkConfig
+): HttpNetworkConfig {
+  const accounts: HttpNetworkAccountsConfig =
+    networkConfig.accounts === undefined
+      ? defaultHttpNetworkParams.accounts
+      : isHdAccountsConfig(networkConfig.accounts)
+      ? {
+          ...defaultHdAccountsConfigParams,
+          ...networkConfig.accounts,
+        }
+      : Array.isArray(networkConfig.accounts)
+      ? networkConfig.accounts.map(normalizeHexString)
+      : "remote";
+
+  const url = networkConfig.url;
+
+  assertHardhatInvariant(
```

Comment:
I actually did that first, but it gets a bit weird because the `localhost` network doesn't require a `url`. https://github.com/nomiclabs/buidler/blob/618299a176ab8a69a50cac88858e7351b027f405/packages/hardhat-core/src/types/config.ts#L5

Reply:
I'm still open to doing it

Reply:
I guess if `url` is mandatory, then `localhost` uses `127.0.0.1:8545` if it's not configured at all (that is, if there's no `localhost` in `networks`), but if you do configure `localhost` you have to configure the URL, right?

I believe it's an acceptable trade-off, but I'm actually fine with either option.